src/models/real-fevr-opener.ts

```diff
@@ -168,8 +166,28 @@ export class OpenerRealFvr extends Model<OpenerRealFvrMethods> implements Deploy
     return this.callTx(this.contract.methods.getRegisteredIDs(_address));
   }
 
+  async getMarketplaceDistributionForERC721(tokenId: number) {
+    const res = await this.callTx(this.contract.methods.getMarketplaceDistributionForERC721(tokenId));
+
+    return {
+      distributionAmounts: res[0].map(function(a: string) { return parseInt(a, 10); }),
+      distributionAddresses: res[1],
+    };
+  }
+
+  async getPurchaseToken() {
+    return this.callTx(this.contract.methods._purchaseToken());
+  }
+
+  async getTokenWorthof1USD() {
+    return fromDecimals(
+      await this.callTx(this.contract.methods._realFvrTokenPriceUSD()),
+      this.erc20 ? this.erc20.decimals : 18,
```

Comment:
logic is the same as the isEthTx but different, stick to one method (I'd use the private prop)

Reply:
done